### PR TITLE
re-add coffee-script@1.9.3; transfer to git autoupdate

### PR DESCRIPTION
**Pull request by humbletim**
Here are regenerated builds of the coffee-script 1.9.3 browser module.

Note that CoffeeScript's npm package _does not_ contain the bundled browser module.  And also that CoffeeScript's git repo _does not_ distribute a regular build of the module (`extras/coffee-script.js` is already minified).

Even though there are still manual steps required, I've updated `ajax/libs/coffee-script/package.json` to autoupdate from the git repo.

The commands I used to generate a pair of non/minified coffeescript browser modules are:

``` sh
git clone git://github.com/jashkenas/coffeescript.git
cd coffeescript
git checkout 1.9.3
cake build:browser  # produces *minified* extras/coffee-script.js
mv extras/coffee-script.js extras/coffee-script.min.js
MINIFY=false cake build:browser  # produces *non-minified* extras/coffee-script.js
```

CC: #4869

### CODE REVIEW

**Comment by PeterDaveHello** [comment]
Wait for +1

**Comment by michaelficarra** [comment]
It'd be nice if @github showed a SHA for truncated files so we could easily verify their contents.

**Comment by PeterDaveHello** [comment]
@michaelficarra do I need sha1 of them? I can do that for you.

**Comment by PeterDaveHello** [comment]
@humbletim sorry for the late reply, seems the upstream only contains the minified version without `.min` in the filename, but I saw you have both src and minified file, would you please confirm that? Thanks.

**Comment by humbletim** [comment]
@PeterDaveHello Correct the upstream maintains a minified version for browsers as `extras/coffee-script.js`

CoffeeScript's [Cakefile](https://github.com/jashkenas/coffeescript/blob/ec333a0/Cakefile#L138-L167) supports an undocumented(?) but trivial environment variable for disabling minification -- which I used alternatively to produce both the src and minified build for this pull request.

Maybe it would be better to call the src version `coffee-script.debug.js`? (to maintain filename parity with minified upstream version?)

**Comment by PeterDaveHello** [comment]
I prefer to use official build.

**Comment by humbletim** [comment]
Fair enough -- now that you have all the facts, what exactly would you like to see included into a pull request?

Also did you notice this one's [package.json](https://github.com/humbletim/cdnjs/blob/5cf03cff/ajax/libs/coffee-script/package.json#L29)?  I believe it conforms to CDNJS autoupdate.md specs and already references the official CoffeeScript repo, along with its singular "official build" artifact.

**Comment by PeterDaveHello** [comment]
> Fair enough -- now that you have all the facts, what exactly would you like to see included into a pull request?

So please make the file you added as the same from upstream: https://github.com/jashkenas/coffeescript/tree/1.9.3/extras


> Also did you notice this one's package.json? I believe it conforms to CDNJS autoupdate.md specs and already references the official CoffeeScript repo, along with its singular "official build" artifact.

Yes, it will work for the missing and future versions.

**Comment by humbletim** [comment]
> So please make the file you added as the same from upstream: https://github.com/jashkenas/coffeescript/tree/1.9.3/extras

OK, cool -- I think this pull request should be good to go now.

Here are some SHA1 checksums:

```sh
$ curl -s https://raw.githubusercontent.com/jashkenas/coffeescript/1.9.3/extras/coffee-script.js | sha1sum
cf79fc1831bdc0c0adc423456ce1d9eeec96703b  -
$ curl -s https://raw.githubusercontent.com/humbletim/cdnjs/master/ajax/libs/coffee-script/1.9.3/coffee-script.js | sha1sum 
cf79fc1831bdc0c0adc423456ce1d9eeec96703b  -
$ curl -s https://raw.githubusercontent.com/humbletim/cdnjs/5b89d59/ajax/libs/coffee-script/1.9.3/coffee-script.js | sha1sum
cf79fc1831bdc0c0adc423456ce1d9eeec96703b  -
```

**Comment by PeterDaveHello** [comment]
@humbletim would you please solve the conflicts so that I can merge it? Or maybe sending a new PR is easier, thanks.

**Comment by humbletim** [comment]
OK... recreated my master from yours, re-applied the change set and cleaned package.json up (so that it passes your latest circleci validation tests).

File: ajax/libs/coffee-script/package.json [hunk 1]
```diff
@@ -12,19 +12,17 @@
...
-        "coffee-script.js"
-      ]
-    }
-  ],
   "repository": {
     "type": "git",
     "url": "https://github.com/jashkenas/coffeescript.git"
   },
-  "license": "MIT",
-  "author": "Jeremy Ashkenas"
```
Comment (PeterDaveHello) [inline ajax/libs/coffee-script/package.json]: Would you please leave license and author info alone? Thanks.

File: ajax/libs/coffee-script/package.json [hunk 2]
```diff
@@ -12,19 +12,17 @@
...
+  "namespace": "CoffeeScript",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/jashkenas/coffeescript.git",
+    "basePath": "/extras",
+    "files": [
+      "coffee-script.js"
```
Comment (PeterDaveHello) [inline ajax/libs/coffee-script/package.json]: Let's use a more flexible pattern  `"coffee-script*.+(js|map)"` here, thanks.

File: ajax/libs/coffee-script/package.json [hunk 3]
```diff
@@ -12,19 +12,17 @@
...
     "type": "git",
     "url": "https://github.com/jashkenas/coffeescript.git"
   },
-  "license": "MIT",
-  "author": "Jeremy Ashkenas"
+  "namespace": "CoffeeScript",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/jashkenas/coffeescript.git",
+    "basePath": "/extras",
```
Comment (PeterDaveHello) [inline ajax/libs/coffee-script/package.json]: In fact, there is no need to have a leading slash.

**Comment by humbletim** [comment]
> Would you please leave license and author info alone? Thanks.

That comment bothers me a little... since no such edits were made on my side.  Instead you seem to have added/updated those properties in master after the original PR (and its package.json) were created last year -- making those lines appear like deletions now when compared.

In any event, I'll take a look at updating the file per your latest feedback over the upcoming US holiday break.

**Comment by PeterDaveHello** [comment]
hmmmm ... so maybe you could just rebase your branch with our latest master branch to prevent this problem, thanks.

**Comment by STRd6** [comment]
Is there anything I can do to move this along? I'd like to have working CoffeeScript versions available from cdnjs.

**Comment by PeterDaveHello** [comment]
@humbletim do you still want to finish this PR, if not, maybe @STRd6 can just send a new one.

**Comment by PeterDaveHello** [comment]
Hello, any updates here? Thanks.

**Comment by humbletim** [comment]
Sorry for the delayed reply -- I'll have to pass on doing this one for now.   Happy to address any questions regarding previous attempts and to help test/confirm someone else's PR.

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by dakshshah96** [comment]
@PeterDaveHello What should we do about this PR?

**Comment by metalim** [comment]
Bump.

Is the issue solved?

**Comment by sufuf3** [comment]
@PeterDaveHello I think we added the auto-update config https://github.com/cdnjs/cdnjs/blob/master/ajax/libs/coffee-script/package.json#L15
Can we close this PR?
Also, the contributor didn't create a new branch.
If we really need to fix this problem, we need to close this PR an send another PR.

Thank you.

**Comment by sufuf3** [comment]
@PeterDaveHello I think we added the auto-update config https://github.com/cdnjs/cdnjs/blob/master/ajax/libs/coffee-script/package.json#L15
Can we close this PR?
Also, the contributor didn't create a new branch.
If we really need to fix this problem, we need to close this PR an send another PR.

Thank you

**Comment by sufuf3** [comment]
I think we solved this https://github.com/cdnjs/cdnjs/blob/master/ajax/libs/coffee-script/package.json#L15.
Should we close this PR?
@PeterDaveHello Would you please take a look?
Many thanks.

**Comment by sashberd** [comment]
@PeterDaveHello This PR is waiting for you review

**Comment by PeterDaveHello** [comment]
Sorry for the late reply, yes, I think we can close this PR, thanks for all the help!

**Pull request closed without merging**